src/ImageSharp/Processing/Processors/Convolution/BokehBlurProcessor.cs

```diff
@@ -0,0 +1,534 @@
+        /// </param>
+        /// <param name="gamma">
+        /// The gamma highlight factor to use to further process the image.
+        /// </param>
+        public BokehBlurProcessor(int radius = 32, int components = 2, float gamma = 3)
+        {
+            this.Radius = radius;
+            this.kernelSize = (radius * 2) + 1;
+            this.componentsCount = components;
+
+            SixLabors.Guard.MustBeGreaterThanOrEqualTo(gamma, 1, nameof(gamma));
+            this.Gamma = gamma;
+
+            // Reuse the initialized values from the cache, if possible
+            if (Cache.TryGetValue((radius, components), out (Vector4[], float, Complex64[][]) info))
+            {
+                this.kernelParameters = info.Item1;
+                this.kernelsScale = info.Item2;
+                this.kernels = info.Item3;
+            }
+            else
+            {
+                // Initialize the complex kernels and parameters with the current arguments
+                (this.kernelParameters, this.kernelsScale) = this.GetParameters();
+
+                this.kernels = new Complex64[this.kernelParameters.Length][];
+                ref Vector4 baseRef = ref MemoryMarshal.GetReference(this.kernelParameters.AsSpan());
+                for (int i = 0; i < this.kernelParameters.Length; i++)
+                {
+                    ref Vector4 paramsRef = ref Unsafe.Add(ref baseRef, i);
+                    this.kernels[i] = this.CreateComplex1DKernel(paramsRef.X, paramsRef.Y);
+                }
+
+                this.NormalizeKernels();
+
+                // Store them in the cache for future use
+                Cache.Add((radius, components), (this.kernelParameters, this.kernelsScale, this.kernels));
+            }
+        }
+
+        /// <summary>
+        /// Gets the Radius
+        /// </summary>
+        public int Radius { get; }
+
+        /// <summary>
+        /// Gets the complex kernels to use to apply the blur for the current instance
+        /// </summary>
+        public IReadOnlyList<Complex64[]> Kernels => this.kernels;
+
+        /// <summary>
+        /// Gets the kernel parameters used to compute the pixel values from each complex pixel
+        /// </summary>
+        public IReadOnlyList<Vector4> KernelParameters => this.kernelParameters;
+
+        /// <summary>
+        /// Gets the gamma highlight factor to use when applying the effect
+        /// </summary>
+        public float Gamma { get; }
+
+        /// <summary>
+        /// Gets the kernel scales to adjust the component values in each kernel
+        /// </summary>
+        private static IReadOnlyList<float> KernelScales { get; } = new[] { 1.4f, 1.2f, 1.2f, 1.2f, 1.2f, 1.2f };
+
+        /// <summary>
+        /// Gets the available bokeh blur kernel parameters
+        /// </summary>
+        private static IReadOnlyList<Vector4[]> KernelComponents { get; } = new[]
```

Comment:
Same as before, this is already readonly. I can make both of these fields instead of properties if you prefer though. The access time wouldn't be a problem either way, as both of these are only accessed once in the constructor, if the kernels aren't present in the cache.